File: OEM/Domain/ExecutedOperations/ActualEnd.cs
namespace OEMAPI.Domain.ExecutedOperations
{
    public class ActualEnd : ValueObject
    {
        public DateTime? Value { get; }

        public ActualEnd(DateTime? value)
        {
            Value = value;
        }

        protected override IEnumerable<object> GetEqualityComponents()
        {
            yield return Value;
        }
    }
}

File: OEM/src/Domain/ExecutedOperations/ActualEnd.ts
import { ValueObject } from "../core/domain/ValueObject";
import { Result } from "../core/logic/Result";

interface ActualEndProps {
  value: Date;
}

export class ActualEnd extends ValueObject<ActualEndProps> {

  get value(): Date {
    return this.props.value;
  }

  private constructor(props: ActualEndProps) {
    super(props);
  }

  public static create(date: Date): Result<ActualEnd> {
    return Result.ok<ActualEnd>(
      new ActualEnd({ value: date })
    );
  }
}
